src/client.ts

```diff
@@ -7143,15 +7146,24 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
      * @param {string} loginType The type of SSO login we are doing (sso or cas).
      *     Defaults to 'sso'.
      * @param {string} idpId The ID of the Identity Provider being targeted, optional.
+     * @param {"login"|"register"} action the SSO flow to indicate to the IdP, optional.
      * @return {string} The HS URL to hit to begin the SSO login process.
      */
-    public getSsoLoginUrl(redirectUrl: string, loginType = "sso", idpId?: string): string {
+    public getSsoLoginUrl(
+        redirectUrl: string,
+        loginType = "sso",
+        idpId?: string,
+        action?: "login" | "register",
+    ): string {
         let url = "/login/" + loginType + "/redirect";
         if (idpId) {
             url += "/" + idpId;
         }
 
-        return this.http.getUrl(url, { redirectUrl }, PREFIX_R0);
+        const params = { redirectUrl };
+        params[SSO_ACTION_PARAM.unstable] = action;
```

Comment:
```suggestion
        const params = { 
            redirectUrl,
            [SSO_ACTION_PARAM.unstable]: action,
        };
```

I think this looks a bit nicer

<hr>

Also, are we ok with sending the server `undefined`s?